src/components/__tests__/HierarchicalMenu.js

```diff
@@ -488,6 +489,28 @@ it('calls the Panel mixin with `items.length`', () => {
   expect(wrapper.vm.mapStateToCanRefine({})).toBe(false);
 });
 
+it('exposes send-event method for insights middleware', () => {
+  const sendEvent = jest.fn();
+  __setState({
+    ...defaultState,
+    sendEvent,
+  });
+
+  const wrapper = mount(HierarchicalMenu, {
+    propsData: defaultProps,
+    scopedSlots: {
+      default: `
+      <div slot-scope="{ sendEvent }">
+        <button @click="sendEvent()">Click</button>
+      </div>
+      `,
+    },
+  });
+
+  wrapper.find('button').trigger('click');
+  expect(sendEvent).toHaveBeenCalledTimes(1);
+});
+
```

Comment:
@Haroenv do you see any better way to test if `sendEvent` exists?
By the way, I think we can only check if it exists. The logic of sending events is already tested in InstantSearch.js.

Reply:
this is a good e2e style test, anything else will depend on internals too much IMO